stats/bookkeeper-stats-providers/prometheus-metrics-provider/src/main/java/org/apache/bookkeeper/stats/prometheus/ThreadScopedDataSketchesStatsLogger.java

```diff
@@ -96,15 +96,25 @@ private DataSketchesOpStatsLogger getStatsLogger() {
             ThreadRegistry.ThreadPoolThread tpt = ThreadRegistry.get();
             if (tpt == null) {
                 statsLoggers.set(defaultStatsLogger);
-                provider.opStats.put(new ScopeContext(scopeContext.getScope(), originalLabels), defaultStatsLogger);
+                DataSketchesOpStatsLogger previous = provider.opStats
+                        .put(new ScopeContext(scopeContext.getScope(), originalLabels), defaultStatsLogger);
+                // If we overwrite a logger, metrics will not be collected correctly
+                if (previous != null && previous != defaultStatsLogger) {
+                    throw new IllegalStateException("Invalid state. Overwrote a stats logger.");
+                }
```

Comment:
I would add a log line about this case, because we should notice when something is using the "default" statslogger, its matrics don't make much sense and they won't probably meaningful for administrators.
So it is fine to not break but we should log a warning

Reply:
> its matrics don't make much sense

Can just remove these labels? It's a slight breaking change, but they seem unnecessary.

Reply:
Also, many metrics were removed in the 4.16 release, implying there isn't a strict sense of compatibility here.